src/main/common/huffman.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "huffman.h"
+
+
+int huffmanEncodeBuf(uint8_t *outBuf, int outBufLen, const uint8_t *inBuf, int inLen, const huffmanTable_t *huffmanTable)
+{
+    int ret = 0;
+
+    uint8_t *outByte = outBuf;
+    *outByte = 0;
+    uint8_t outBit = 0x80;
+    for (int ii = 0; ii< inLen; ++ii) {
```

Comment:
I follow the "get it working" and then "get it working well" approach. So optimisation can come later.

But if you want to make a PR I'll be happy to take it.